jans-cedarling/cedarling/src/common/policy_store.rs

```diff
@@ -5,155 +5,430 @@
+impl<'de> Deserialize<'de> for TokenKind {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
     where
         D: serde::Deserializer<'de>,
     {
-        let policies =
-            <HashMap<String, RawPolicy> as serde::Deserialize>::deserialize(deserializer)?;
-
-        let policy_vec = policies
-            .into_iter()
-            .map(|(id, policy_raw)| {
-                let policy = parse_policy::<D>(&id, policy_raw).map_err(|err| {
-                    serde::de::Error::custom(format!(
-                        "unable to decode policy with id: {id}, error: {err}"
-                    ))
-                })?;
-                Ok(policy)
-            })
-            .collect::<Result<Vec<cedar_policy::Policy>, _>>()?;
-
-        cedar_policy::PolicySet::from_policies(policy_vec).map_err(|err| {
-            serde::de::Error::custom(format!("{}: {err}", ParsePolicySetMessage::CreatePolicySet))
-        })
+        struct TokenKindVisitor;
+
+        impl<'de> Visitor<'de> for TokenKindVisitor {
+            type Value = TokenKind;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("a valid token kind string")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                match value.to_lowercase().as_str() {
+                    "id" => Ok(TokenKind::Id),
+                    "userinfo" => Ok(TokenKind::Userinfo),
+                    "access" => Ok(TokenKind::Access),
+                    "transaction" => Ok(TokenKind::Transaction),
+                    _ => Err(serde::de::Error::unknown_variant(
+                        &value,
+                        &[
+                            "access_token",
+                            "id_token",
+                            "userinfo_token",
+                            "transaction_token",
+                        ],
+                    )),
+                }
+            }
+        }
+
+        deserializer.deserialize_string(TokenKindVisitor)
     }
+}
 
-    // function to deserialize a single policy from `PolicyRaw`
-    fn parse_policy<'de, D>(
-        id: &str,
-        policy_raw: RawPolicy,
-    ) -> Result<cedar_policy::Policy, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
-        let decoded = BASE64_STANDARD
-            .decode(policy_raw.policy_content.as_str())
-            .map_err(|err| {
-                serde::de::Error::custom(format!("{}: {err}", ParsePolicySetMessage::Base64))
+/// Validates the `cedar_version` field.
+///
+/// This function checks that the version string follows the format `major.minor.patch`,
+/// where each component is a valid number. This also supports having a "v" prefix in the
+/// version, e.g. `v1.0.1`.
+fn check_cedar_version<'de, D>(deserializer: D) -> Result<String, D::Error>
```

Comment:
we can return here structure with 3 fields major, minor, patch

Reply:
fixed in [f07fcc2](https://github.com/JanssenProject/jans/pull/9910/commits/f07fcc2e45d6cb0e4e638f77ebf7f51632659042). returns `semver::Version` now.